mods/persistence/modules/trade_beacons/trade_beacons.dm

```diff
@@ -0,0 +1,103 @@
+/atom/proc/get_adjacent_trade_beacons()
+	var/list/adjacent_trade_beacons = list()
+	var/atom/target = src
+	if(!istype(src, /turf))
+		target = get_turf(src)
+	var/obj/ob = target.get_owning_overmap_object()
+	if(ob)
+		var/turf/T = get_turf(ob)
+		if(T)
+			for(var/obj/beacon in SStrade_beacons.all_trade_beacons)
+				var/turf/beacon_turf = get_turf(beacon)
+				if(beacon_turf && beacon_turf.Adjacent(T))
+					adjacent_trade_beacons |= beacon
+	return adjacent_trade_beacons
+
+/obj/effect/overmap/trade_beacon
+	name = "Trade Beacon"
+	var/list/possible_imports = list() // structure is list(/datum/beacon_import/example = 100) list(type = percentage chance of appearing).
+	var/list/possible_exports = list()
+	var/list/active_imports = list()
+	var/list/active_exports = list()
+	icon = 'icons/obj/supplybeacon.dmi'
+	icon_state = "beacon_active"
+	var/datum/extension/network_device/trade_controller/linked_controller
+	var/datum/money_account/beacon_account
+	var/start_x = 0
+	var/start_y = 0
+
+/obj/effect/overmap/trade_beacon/proc/move_to_starting_location()
+	var/datum/overmap/overmap = global.overmaps_by_name[overmap_id]
+	var/location
+
+	if(start_x && start_y)
+		location = locate(start_x, start_y, overmap.assigned_z)
+	else
+		var/list/candidate_turfs = block(
+		locate(OVERMAP_EDGE, OVERMAP_EDGE, overmap.assigned_z),
+		locate(overmap.map_size_x - OVERMAP_EDGE, overmap.map_size_y - OVERMAP_EDGE, overmap.assigned_z)
+		)
+
+		candidate_turfs = where(candidate_turfs, /proc/can_not_locate, /obj/effect/overmap)
+		location = SAFEPICK(candidate_turfs) || locate(
+			rand(OVERMAP_EDGE, overmap.map_size_x - OVERMAP_EDGE),
+			rand(OVERMAP_EDGE, overmap.map_size_y - OVERMAP_EDGE),
+			overmap.assigned_z
+		)
+
+	forceMove(location)
+
+
+
+/obj/effect/overmap/trade_beacon/New()
+	beacon_account = new()
+	beacon_account.account_name = name
+
+/obj/effect/overmap/trade_beacon/Initialize()
+	..()
```

Comment:
You're missing the return value. Unit tests will complain.
```suggestion
	. = ..()
```

Reply:
fixed